src/LiveComponent/src/Hydration/DoctrineEntityHydrationExtension.php

```diff
@@ -81,14 +81,44 @@ public function dehydrate(object $object): mixed
 
     private function objectManagerFor(string $class): ?ObjectManager
     {
-        if (!class_exists($class)) {
+        if (class_exists($class)) {
+            // Keep the standard way for a class
+            // todo cache/warmup an array of classes that are "doctrine objects"
+            foreach ($this->managerRegistries as $registry) {
+                if ($om = $registry->getManagerForClass($class)) {
+                    return self::ensureManagedObject($om, $class);
+                }
+            }
+
             return null;
         }
 
-        // todo cache/warmup an array of classes that are "doctrine objects"
-        foreach ($this->managerRegistries as $registry) {
-            if ($om = $registry->getManagerForClass($class)) {
-                return self::ensureManagedObject($om, $class);
+        if (interface_exists($class)) {
+            // special handler for interfaces
+            // For use cases : @see https://symfony.com/doc/current/doctrine/resolve_target_entity.html
+            // As today, getManagerForClass don't resolve nicely aliased interfaces
+            // The workaround is to enum over each object manager and trying to get metadata
+            // The metadata are indeed, resolved nicely
+            // Fore more details :
```

Comment:
Can we turn this into two or three lines, insisting more on what we "do" than on what "was" the problem ?